src/Extension/FluentExtension.php

```diff
@@ -707,20 +704,18 @@ static function (FluentState $state) use ($owner, $locale): ?DataObject {
                 $duplicate = $original->duplicate();
                 // Attach the duplicated relation to localised record
                 $localisedRecord->setComponent($relation, $duplicate);
-
-                // Store the record so we can write it later, this makes us execute only one write
-                // even though we duplicate several relations
-                $objectsToWrite[$locale->Locale] = $localisedRecord;
+                $hasRecordsToWrite = true;
             }
-        }
 
-        foreach ($objectsToWrite as $localeCode => $localisedRecord) {
-            FluentState::singleton()->withState(
-                static function (FluentState $state) use ($localeCode, $localisedRecord): void {
-                    $state->setLocale($localeCode);
-                    $localisedRecord->write();
-                }
-            );
+            // Update localised record if any localised copies took place
+            if ($hasRecordsToWrite) {
```

Comment:
[nitpick] Would prefer early exit approach here to reduce code nesting.

Reply:
you mean 

```php
if (!$hasRecordsToWrite) {
  continue
}
FluentState::singleton()->withState(
    static function (FluentState $state) use ($recordLocale, $localisedRecord): void {
        $state->setLocale($recordLocale->getLocale());
        $localisedRecord->write();
    }
);
```

Reply:
That's correct ;-)

Reply:
I'll update when I got some time :D